src/app/components/home/home.component.ts

```diff
@@ -21,7 +21,9 @@ export class HomeComponent implements OnInit {
 
   loadCoupons(): void {
     this.generalService.loadCoupons().subscribe((values: Coupon[]) => {
-      this.coupons = values;
+      this.coupons = values.filter((value: Coupon) => {
+        return (new Date(value.startDate).valueOf()) <= new Date().valueOf();
```

Comment:
in this line, does it return only if the ""  <= new Date().valueOf();"" condition is met?
does it not return if new date(valueofBlahblah) > new Date(.valueOf)?